mnemosyne/migrations/e7_311_tables.py

```
@@ -0,0 +1,125 @@
+migration should be updated to match.
+
+Safe to re-run (idempotent — uses ``CREATE TABLE IF NOT EXISTS``
+and try/except for indices, matching the upstream behavior). The
+migration does not delete any data or drop any tables.
+"""
+
+from __future__ import annotations
+
+import sqlite3
+from pathlib import Path
+from typing import Dict, List
```

Ruff (0.15.20)

[warning] `typing.Dict` is deprecated, use `dict` instead

(UP035)